components/esp32_ble_presense/esp32_ble_presense.cpp

```diff
@@ -12,6 +12,14 @@
 #define bleScanInterval 0x80 // Used to determine antenna sharing between Bluetooth and WiFi. Do not modify unless you are confident you know what you're doing
 #define bleScanWindow 0x40 // Used to determine antenna sharing between Bluetooth and WiFi. Do not modify unless you are confident you know what you're doing
 
+// Undo the default loose definitions in our file only
+#pragma GCC diagnostic error "-Wdeprecated-declarations"
+#pragma GCC diagnostic error "-Wsign-compare"
+#pragma GCC diagnostic error "-Wunused-but-set-variable"
+#pragma GCC diagnostic error "-Wunused-function"
+#pragma GCC diagnostic error "-Wunused-parameter"
+#pragma GCC diagnostic error "-Wunused-variable"
```

Comment:
I'm going to rework this one - its not really ready to merge - if you disable all logging you'll probably get errors - I will fix that and reissue.

Reply:
Actually after investigation this is only affecting later code (I'm busy working on more surprises)

Reply:
I've added a test to build with and without logging